src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -384,7 +385,13 @@ private static bool TryResolveStateMachineMethod(ref MethodBase method, out Type
                 return false;
             }
 
-            MethodInfo[]? methods = parentType.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
+                Justification = "Using Reflection to find the state machine's corresponding method is safe because the corresponding method is the only " +
+                                "caller of the state machine. If the state machine is present, the corresponding method will be too.")]
+            static MethodInfo[]? GetDeclaredMethods(Type type) =>
```

Comment:
You separated this into a separate method just to keep the suppression scoped to be as small as possible?

Reply:
Correct. That way if other reflection gets added to the method in the future, the ILLinker will warn about other usages.